pycoal/mineral.py

```diff
@@ -85,49 +121,31 @@ def SAM(image_file_name, classified_file_name, library_file_name, scores_file_na
     if scores_file_name is not None:
         # allocate a zero-initialized MxN array for the scores image
         scored = numpy.zeros(shape=(M,N), dtype=numpy.float64)
+        scoredJobLib = numpy.zeros(shape=(M,N), dtype=numpy.float64)
+       
+    num_cores = multiprocessing.cpu_count()
+    
 
-    # for each pixel in the image
-    for x in range(M):
-
-        for y in range(N):
-
-            # read the pixel from the file
-            pixel = data[x,y]
-
-            # if it is not a no data pixel
-            if not numpy.isclose(pixel[0], -0.005) and not pixel[0]==-50:
-
-                # resample the pixel ignoring NaNs from target bands that don't overlap
-                # TODO fix spectral library so that bands are in order
-                resampled_pixel = numpy.nan_to_num(resample(pixel))
-
-                # calculate spectral angles
-                angles = spectral.spectral_angles(resampled_pixel[numpy.newaxis,
-                                                                 numpy.newaxis,
-                                                                 ...],
-                                                  library.spectra)
-
-                # normalize confidence values from [pi,0] to [0,1]
-                for z in range(angles.shape[2]):
-                    angles[0,0,z] = 1-angles[0,0,z]/math.pi
-
-                # get index of class with largest confidence value
-                index_of_max = numpy.argmax(angles)
-
-                # get confidence value of the classied pixel
-                score = angles[0,0,index_of_max]
 
-                # classify pixel if confidence above threshold
-                if score > threshold:
+    start = time.time()
+    scoredSingle = numpy.array(Parallel(n_jobs=num_cores)(delayed(CalculatePixel)(data[x,y], 
+                        x, y, classified, library, threshold, resample, scores_file_name)
+                        for x in range(M) for y in range(N)), dtype=numpy.float64)
 
-                    # index from one (after zero for no data)
-                    classified[x,y] = index_of_max + 1
+    #puts it all in one single array, need to make 2d array
+    k = 0
+    for i in range(M):
+        for j in range(N):
+            if scoredSingle[k] is scoredSingle[k]:
+                scoredJobLib[i][j] = scoredSingle[k]
+            k+=1
+    
+    end = time.time()
+    seconds = end - start
+   
 
-                    if scores_file_name is not None:
-                        # store score value
-                        scored[x,y] = score
 
-    # save the classified image to a file
+    print("Time to run: {} seconds".format(seconds))
```

Comment:
Please redirect to logging. An example can be seen [here](https://github.com/capstone-coal/pycoal/blob/master/pycoal/mineral.py#L87).